libsupport/src/EntityTypeManager.cpp

```diff
@@ -1,5 +1,40 @@
 #include "katana/EntityTypeManager.h"
 
+#include "katana/Logging.h"
+#include "katana/Result.h"
+
+size_t
+katana::EntityTypeManager::CalculateSetOfEntityTypeIDsSize(
+    EntityTypeID max_id) {
+  // min number of bits to fit bitset[max_id] is max_id +1
+  size_t min_size = max_id + 1;
+
+  KATANA_LOG_VASSERT(
+      max_id < kInvalidEntityType, "Katana only supports {} entity types",
+      kInvalidEntityType);
+
+  size_t new_size = kDefaultSetOfEntityTypeIDsSize;
```

Comment:
Why do we start with `kDefaultSetOfEntityTypeIDsSize` and not the actual current size?

Reply:
We don't really care about the current size, we really only care about how big we need the set to be.
I could see the argument for passing in the current size to ensure the calculated size is larger than the current size, but at this point that is on the caller. 
This function is really only for calculating how bit of a set we should use to hold some arbitrary number of EntityTypeIDs